### DISCUSSION TITLE
The bot fails with "filename argument required" on yarn 3 monorepo

### DISCUSSION THREAD

**Discussion by dmitriy-kudelko** [renovatebot/renovate#23389] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Hello, dear Renovate maintainers!

We're trying to migrate one of our repos to yarn-based monorepo. For this purpose we used latest yarn 3.6.1 with `nodeLinker: node-modules` and zero installs enabled (all deps are stored in archives under `.yarn/cache` folder).

Another, possibly important detail, is that we have to use Node.js 14, since it's required by one of our dependencies.

A problem however is that after migrating to monorepo all renovate bot PRs now fail with the same error (see the logs).

I've created a minimal reproduction repo here: https://github.com/dmitriy-kudelko/flex-plugins/pull/3

Could you please have a look what might be wrong?

Interestingly, a somewhat similar error happens when trying to validate the config locally via `renovate-config-validator`:

<details>
<summary>Validator logs</summary>

```
RegEx ??= RegExp;
      ^^^

SyntaxError: Unexpected token '??='
    at wrapSafe (internal/modules/cjs/loader.js:1029:16)
    at Module._compile (internal/modules/cjs/loader.js:1078:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1143:10)
    at Module.load (internal/modules/cjs/loader.js:979:32)
    at Function.Module._load (internal/modules/cjs/loader.js:819:12)
    at Module.require (internal/modules/cjs/loader.js:1003:19)
    at require (internal/modules/cjs/helpers.js:107:18)
    at Object.<anonymous> (/Users/gemini/.nvm/versions/node/v14.21.3/lib/node_modules/renovate/lib/util/url.ts:6:1)
    at Module._compile (internal/modules/cjs/loader.js:1114:14)
```
</details>

And it doesn't occur when switching to Node.js 18 locally. So these might be somehow related.

Our previous monolithic repo also relied on Node.js 14, and renovate bot worked just fine. So I assume it's something related specifically to how yarn repos with older Node.js versions are handled

### Relevant debug logs

<details><summary>Logs</summary>

```
/usr/local/bin/docker: line 4: .: filename argument required
.: usage: . filename [arguments]
/opt/containerbase/tools/corepack/0.19.0/node_modules/corepack/dist/lib/corepack.cjs:39787
      process.exitCode ??= code;
                       ^^^

SyntaxError: Unexpected token '??='
    at wrapSafe (internal/modules/cjs/loader.js:1029:16)
    at Module._compile (internal/modules/cjs/loader.js:1078:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1143:10)
    at Module.load (internal/modules/cjs/loader.js:979:32)
    at Function.Module._load (internal/modules/cjs/loader.js:819:12)
    at Module.require (internal/modules/cjs/loader.js:1003:19)
    at require (internal/modules/cjs/helpers.js:107:18)
    at Object.<anonymous> (/opt/containerbase/tools/corepack/0.19.0/node_modules/corepack/dist/corepack.js:2:1)
    at Module._compile (internal/modules/cjs/loader.js:1114:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1143:10)

```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
Corepack not supported in node 14?

**Reply by dmitriy-kudelko** [reply]
@rarkins , I assume it does: https://nodejs.org/tr/blog/release/v14.19.0#corepack

**Reply by rarkins** [reply]
https://github.com/nodejs/corepack/releases/tag/v0.18.0

**Reply by dmitriy-kudelko** [reply]
@rarkins , yeah it explains things..

Locally it still works though since I have a previous version installed:

```sh
$ corepack -v
0.15.1
```

Any way to work-around it with renovate? We absolutely cannot migrate to Node > 14 as one of our core dependencies still rely on it.
We're working on some custom Twilio Flex plugins and its CLI isn't yet Node 16-ready: https://www.twilio.com/docs/flex/developer/plugins/cli/install#prerequisites.

Thanks for the quick response BTW. Really appreciate this 🤝

**Reply by rarkins** [reply]
We could support checking constraints>corepack here: https://github.com/renovatebot/renovate/blob/a1b8960a665037d5e67756946aff83f0f867e4f8/lib/modules/manager/npm/post-update/yarn.ts#L127

But I think you'd need to submit the PR if so

**Reply by viceice** [reply]
workaround:

add a corepack constraint with value `^0.18.0` to the renovate config

**Reply by viceice** [reply]
@rarkins we should probably find a way to pass the resolved node version as strict constraints to any npm tool we install with containerbase. that way it should work for most people without any issues.

**Reply by rarkins** [reply]
Does corepack constraint work already?

**Reply by viceice** [reply]
don't know 🤷‍♂️

**Reply by rarkins** [reply]
I think it needs this: https://github.com/renovatebot/renovate/compare/main...fix/corepack-constraint?quick_pull=1

**Reply by dmitriy-kudelko** [reply]
So guys, do you want me to open this PR? I just planned looking at it but, it turned out the fix is already there :)
Or do you still need to discuss if this solution is a proper one?

**Reply by rarkins** [reply]
I'm on vacation and on mobile so wont be continuing it- you're welcome to take my code and add a test in a PR!

**Reply by dmitriy-kudelko** [reply]
Alright, sounds like a plan! Will do.

**Reply by dmitriy-kudelko** [reply, marked as answer]
@rarkins , @viceice , here it is guys: https://github.com/renovatebot/renovate/pull/23395

Thank you for all your assistance! I'm gonna be waiting for review.
